src/v/kafka/server/materialized_partition.h

```diff
@@ -51,13 +51,13 @@ class materialized_partition final : public kafka::partition_proxy::impl {
     ss::future<model::record_batch_reader> make_reader(
       storage::log_reader_config cfg,
       std::optional<model::timeout_clock::time_point>) final {
-        return _log.make_reader(cfg);
+        return _partition->make_reader(cfg);
     }
 
     ss::future<std::optional<storage::timequery_result>>
     timequery(model::timestamp ts, ss::io_priority_class io_pc) final {
-        storage::timequery_config cfg(ts, _log.offsets().dirty_offset, io_pc);
-        return _log.timequery(cfg);
+        storage::timequery_config cfg(ts, last_stable_offset(), io_pc);
```

Comment:
why does dirty offset change to stable offset?

Reply:
`last_stable_offset()` of a `materialized_partition` boils down to the dirty offset